### ISSUE TITLE
Enabling --listen-addr /ip4/0.0.0.0/tcp/9944/ws breaks polkadot.js [via nginx proxy]

### ISSUE THREAD

**Issue by dcolley** [paritytech/polkadot#6571]
Following from #6208 , there is a recommendation [requirement?] to run polkadot with `--listen-addr=/ip4/0.0.0.0/tcp/30334/ws flag.`

I am trying to get my boot nodes to also serve polkadot.js also.

The boot node runs in `docker compose` as follows:

**docker-compose-westend-boot.yml**
```yml
services:

  westend-boot:
    container_name: westend-boot-1
    image: parity/polkadot
    restart: on-failure
    ports:
      - "33012:33012" # p2p port
      - "33013:33013" # prometheus port
      - "33014:33014" # JSON-RPC HTTP server
      - "33015:33015" # JSON-RPC WS server
    volumes:
      - /var/westend/boot-1:/data
    command: [
      "--chain", "westend",
      # "--validator", # not for RPC node!
      "--name", "METASPAN-BOOT",
      "--base-path", "/data",
      "--pruning", "10",
      # "--port", "33012", # p2p protocol TCP port, default 30333
      "--listen-addr", "/ip4/0.0.0.0/tcp/33012",
      "--public-addr", "/ip4/195.144.22.133/tcp/33012",
      "--prometheus-port", "33013",
      "--prometheus-external",
      # HTTP RPC port
      "--rpc-external",
      "--rpc-port", "33014", # HTTP RPC server TCP port
      "--rpc-methods", "safe",
      "--rpc-cors", "all",
      # WS for polkadot.js etc
      "--ws-external",
      "--ws-port", "33015", # WebSockets RPC server TCP port
      # =============================================
      # this next line breaks polkadot.js wss:// config...!
      # but is needed for command line sync...!
      # "--listen-addr", "/ip4/0.0.0.0/tcp/33015/ws",
      # =============================================
      "--public-addr", "/ip4/195.144.22.133/tcp/33015/ws",
      # RPC node
    ]
```
Adding `"--listen-addr", "/ip4/0.0.0.0/tcp/33015/ws",` seems to enable command-line sync as per #6208.

However, that line seems to break 
https://polkadot.js.org/apps/?rpc=wss://ibp-rpc.metaspan.io/kusama#/accounts

Are these services mutually exclusive?

For reference, here is the nginx config:
```conf
upstream rpc_ws_westend {
  server 192.168.1.10:33015;
}
server {
  # external wss://ibp-rpc.metaspan.io/${chain} should arrive here...
  # -> ___.___._.__:443 router external
  # -> 192.168.1.10:443 docker external
  # -> 0.0.0.0:443       docker internal
  listen *:443 ssl;
  server_name ibp-rpc.metaspan.io;
  ssl_certificate     /etc/letsencrypt/live/ibp-rpc.metaspan.io/fullchain.pem; 
  ssl_certificate_key /etc/letsencrypt/live/ibp-rpc.metaspan.io/privkey.pem;

  location /westend {
    proxy_pass http://rpc_ws_westend;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade; # "Upgrade";
    proxy_set_header Host $host;
  }
}
```

**Comment by bkchr** [comment]
Polkadotjs isn't using the libp2p ports to talk to the node. It uses the RPC ports. `--ws-port` is what you are looking for when you want to change the websocket port of the RPC server.

**Comment by dcolley** [comment]
The pjs talks wss to port 443 => nginx
nginx proxy's that to port 33015 => node

pjs/nginx/node works fine without `--listen-addr /ip4/0.0.0.0/tcp/33015/ws`.
But the ws/p2p protocol seems to require it

**Comment by bkchr** [comment]
You are mixing there stuff. Pjs is not connecting to the P2p port. Pjs is using rpcs and this port is controlled using --ws-port. The P2p wss is required for light clients currently.

Tldr, create two port mappings in nginx. One for the rpc port and one for the P2p port.
